nidx/src/searcher/sync.rs

```diff
@@ -39,11 +39,18 @@ pub async fn run_sync(
 ) -> anyhow::Result<()> {
     let mut last_updated_at = PrimitiveDateTime::MIN.replace_year(2000)?;
     loop {
+        let deleted = Index::marked_to_delete(&meta.pool).await?;
         let indexes = Index::recently_updated(&meta.pool, last_updated_at).await?;
         for index in indexes {
             // TODO: Handle errors
+
+            if deleted.contains(&index.id) {
```

Comment:
This should be outside the loop. If an index is deleted that has not been recently updated, it won't be deleted from sync here.